module3/lessons/Week2/RestfulRoutes.md

```diff
@@ -0,0 +1,174 @@
+    </ul>
+</details>
+
+<details>
+    <summary>
+    What is a URI?
+    </summary>
+
+    URI stands for **U**niform **R**esource **I**dentifier. It is the part of the URL after the domain. For our purpose, the URI is the same as our route.
+    <br><br>
+    For example, for the URL <code>https://en.wikipedia.org/wiki/Uniform_Resource_Identifier</code>:
+    <ul>
+        <li> the domain is <code>en.wikipedia.org</code> </li>
+        <li> the URI is <code>/wiki/Uniform_Resource_Identifier</code></li>
+    </ul>
+</details>
+
+## The List
+
+Here is a table of the list of RESTful routes. The given URI is based on the context from our `MvcMovie` application.
+
+| Route Name | URI | HTTP Method | CRUD | Return Type | Purpose |
+|--|--|--|--|--|--|
+| **Index** | `/movies` | GET | Read | View | Display list of all movies |
+| **New** | `/movies/new` | GET | Create | View | Use form to create new movie |
+| **Create** | `/movies` | POST | Create | Redirect | Create new movie in database |
+| **Show** | `/movies/:id` | GET | Read | View | Display details for one specific movie |
+| **Edit** | `/movies/:id/edit` | GET | Update | View | Use form to edit existing movie |
+| **Update** | `/movies/:id` | PUT* | Update | Redirect | Update existing movie in database |
+| **Destroy** | `/movies/:id` | DELETE | Delete | Redirect | Delete existing movie |
+
+* The **Update** route can also use the *PATCH* method. PUT updates an entire resource; PATCH updates a portion of the resource.
+
+## RESTful Routes
+We will focus on two methods in our controller as it pertains to REST.
+
+### Index() method
+
+```c#
+// GET: /Movies
+public IActionResult Index()
+{
+    var movies = _context.Movies;
+    return View(movies);
+}
+```
+
+This is the existing method in our controller from the MVC lesson. As we witnessed previously, this method returns a View that displays all the movies in our database.
+
+### Show(int id)
+
+The Show() method takes an integer and returns the Movie record with that ID number from the database. 
+To accomplish this, we need to do the following:
+* Create a local variable that stores the movie from the `context` object.
+* Return a View to display the details of that movie.
+
+Add this code to `MoviesController.cs` file, preferably after the `Index()` method:
+
+```c#
+// GET: /Movies/<id>
+public IActionResult Show(int id)
+{
+    var movie = _context.Movies.Find(id);
+    return View(movie);
+}
+```
+
+### Show View
+
+We have a View for our Index route that displays all the movies in our database: `Index.cshtml`. 
+We will create a View for our Show route that displays a single movie.
+
+From Solution Explorer:
+* Expand the Views folder
+* Right-click the Movies folder
+* Add > New File
+* Name the new file `Show.cshtml`
+
+Add the following code to the `Show.cshtml` file:
+
+```html
+@model Movie
```

Comment:
We might want to say a little bit of something about why we have @model Movie at the top and how the @ symbol is razor syntax for letting us write C# code in a razor file.

Reply:
Done! I'd like your feedback for clarity and accuracy.